### PR TITLE
feat(storage): add metrics for compaction result check

### PR DESCRIPTION
**Pull request by Little-Wallace**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/storage/src/monitor/compactor_metrics.rs [hunk 1]
```diff
@@ -250,7 +251,13 @@ impl CompactorMetrics {
         );
         let compaction_event_loop_iteration_latency =
             register_histogram_with_registry!(opts, registry).unwrap();
-
+        let compaction_result_check_count = register_int_counter_vec_with_registry!(
+            "storage_compaction_result_check_count",
+            "KBs read from current level during history compactions to next level",
```
Comment (hzxa21) [inline src/storage/src/monitor/compactor_metrics.rs:256]: The description should be modified accordingly.

File: src/storage/src/hummock/compactor/mod.rs [hunk 2]
```diff
@@ -504,6 +504,7 @@ pub fn start_compactor(
                                             tracing::warn!(error = %e.as_report(), "Failed to report task {task_id:?}");
                                         }
                                         if enable_check_compaction_result && need_check_task {
+                                            context.compactor_metrics.compaction_result_check_count.with_label_values(&["compaction"]).inc();
                                             match check_compaction_result(&compact_task, context.clone()).await {
                                                 Err(e) => {
                                                     tracing::warn!(error = %e.as_report(), "Failed to check compaction task {}",compact_task.task_id);
```
Comment (hzxa21) [inline src/storage/src/hummock/compactor/mod.rs:510]: How about adding a metric to record failure happen in the check?

Comment (hzxa21) [reply src/storage/src/hummock/compactor/mod.rs:510]: Same in shared_buffer_compact.rs L269

**Comment by Little-Wallace** [comment]
This problem has been solved by other PR

**Pull request closed without merging**